src/vm/amd64/JitHelpers_InlineGetThread.asm

```diff
@@ -147,6 +147,11 @@ align 16
         jmp     JIT_Box
 NESTED_END JIT_BoxFastMP_InlineGetThread, _TEXT
 
+RELATIVE_PTR_LOAD macro Reg, BaseReg, Offset
```

Comment:
I'm currently working on this, and plan to open pull request for the change early next week.

- the new `TemplateMT` argument is added to `JIT_NewArr1` and other array allocation helpers. It seems that the `ArrayTypeDesc` argument can't be removed, because `JIT_NewArr1` uses it, and other helpers pass it to the `JIT_NewArr1`;
- the argument is embedded to JIT code as constant value, which is figured out during import, in case no runtime lookup is required for the `ArrayTypeDesc`;
- in cases, which require runtime lookup of the value, the `TemplateMT` value is also figured out in runtime by loading it from `ArrayTypeDesc` in JIT code

Could you, please, check whether the described direction is right?